eZ/Publish/API/Repository/Tests/SearchServiceFulltextEmbedTest.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository\Tests;
+
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchHit;
+use eZ\Publish\Core\FieldType\RichText\Value as RichTextValue;
+use eZ\Publish\Core\Repository\Values\Content\Content;
+use eZ\Publish\Core\Repository\Values\Content\ContentCreateStruct;
+
+/**
+ * Test case for full text search in the SearchService (for embed).
+ *
+ * @see \eZ\Publish\API\Repository\SearchService
+ * @group integration
+ * @group search
+ * @group fulltext
+ */
+class SearchServiceFulltextEmbedTest extends BaseTest
+{
+    private const SLAVE_ARTICLE_NAME = 'test1';
+
+    private static $createdIds = [];
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $repository = $this->getRepository(false);
+
+        if (false === $repository->getSearchService()->supports(SearchService::CAPABILITY_ADVANCED_FULLTEXT)) {
+            $this->markTestSkipped('Engine says it does not support advance fulltext format');
+        }
+    }
+
+    public function testPrepareContent(): void
+    {
+        $contentService = $this->getRepository()->getContentService();
+        $baseArticleStruct = $this->prepareBaseArticleStruct();
+
+        $slaveArticleStruct = $this->fillSlaveArticleStruct(clone $baseArticleStruct);
```

Comment:
Kinda not following naming. What does slave mean in this context?

Reply:
Changes, I hope it fits better now